core/src/main/java/io/grpc/transport/inprocess/InProcessTransport.java

```diff
@@ -137,18 +138,18 @@ public synchronized void shutdown() {
       return;
     }
     shutdownStatus = Status.UNAVAILABLE.withDescription("transport was requested to shut down");
-    notifyShutdown();
+    notifyShutdown(shutdownStatus);
```

Comment:
Note that this is a "success" case as far as retry is concerned, but as we discussed, doesn't really matter much because the transport won't call `newStream()` any more. But this does point out that we need to be more clear when to pass Status.OK instead of "the reason for shutdown".